### PR TITLE
Add TypeScript

### PR DESCRIPTION
**Pull request by Darth-Knoppix**
Add TypeScript as per discussion amongst Consumer Team and Frontend Guild

## Why?
- To improve trust in the codebase
- Make refactoring less risky
- Improve DX

### CODE REVIEW

**Review by wizardels — APPROVED** [review]
😍

File: languages/JavaScript/index.md [hunk 1]
```diff
@@ -6,6 +6,27 @@ This document doesn't cover purely preferential things that are already enforced
...
+
+The current website frontend is mostly JavaScript but whenever possible, new files should be written in TypeScript.
+
+### Why TypeScript?
+
+Static type checking is a feature which allows teams to refactor and move quickly with confidence. It helps with onboarding new engineers, improves developer experience and is another layer of trust within the codebase.
+
+### The process of migration
+
+1. **New** files should created as TypeScript unless there is good reason not to.
```
Comment (martyphee) [inline languages/JavaScript/index.md:19]: > files should created

_files should be created_

Comment (shoshana123) [inline languages/JavaScript/index.md:19]: I think we might be missing a `be` in `files should be created`

**Review by martyphee — APPROVED** [review]
Minor comment

**Review by shoshana123 — APPROVED** [review]
This is exciting!

**Review by GBrady-Z — APPROVED** [review]

**Review by lorine-rosenberg — APPROVED** [review]
yes typescript!!!

File: languages/JavaScript/index.md (hunk 1 above)
Comment (shoshana123) [reply languages/JavaScript/index.md:19]: 2 minds think alike 😄

**Comment by Darth-Knoppix** [comment]
Thanks @martyphee and @shoshana123, my excitement got the better of me 😂

**Pull request merged**